test/unit/puppetlabs/services/jruby/jruby_testutils.clj

```diff
@@ -29,6 +28,26 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; JRubyPuppet Test util functions
 
+(schema/defn ^:always-validate
+create-mock-pool-instance :- JRubyInstance
```

Comment:
Huh, weird.  If you've already pored that much time into this, I definitely don't think it's worth pursuing further.  Carry on!